app/views/products/index.html.erb

```diff
@@ -1,32 +1,83 @@
-      </thead>
-      <tbody>
-        <% @products.each do |product| %>
-          <tr class="hover:bg-gray-50">
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800 font-medium"><%= product.name %></td>
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800"><%= product.category&.name || 'N/A' %></td>
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800">$<%= product.price %></td>
-            <td class="px-6 py-4 border-b border-gray-200 text-sm text-gray-800">
-              <% if user_signed_in? %>
-                      <%= button_to 'Add to Cart', add_to_cart_order_path(current_cart, product_id: product.id), method: :post, class: 'bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded' %>
+<% if params[:category].present? %>
+  <% category = params[:category].to_i %>
+  <div class="bg-white">
+    <div class="mx-auto max-w-7xl px-4 py-16 sm:px-6 lg:px-8">
+      <div class="mt-6 grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-4">
+        <% Product.where(category_id: category).each do |product| %>
+          <div class="group relative border border-gray-200 rounded-lg shadow-md hover:shadow-lg transition-shadow bg-white">
+            <div class="overflow-hidden rounded-t-lg bg-gray-100">
+              <% if product.images.attached? %>
+                <%= image_tag product.images.first, alt: product.name, class: "h-48 w-full object-cover object-center" %>
               <% else %>
-                <p class="text-red-500">Login to add to cart</p>
-              <% end %> 
-            </td>
-          </tr>
+                <div class="h-48 flex items-center justify-center text-gray-500 text-sm">
+                  No Image Available
+                </div>
+              <% end %>
+            </div>
+          
+            <div class="p-4">
+              <h3 class="text-sm font-medium text-gray-700 truncate">
+                <a href="#">
+                  <span aria-hidden="true" class="absolute inset-0"></span>
+                  <%= product.name %>
+                </a>
+              </h3>
+              <p class="mt-2 text-sm text-gray-500"><%= truncate(product.description, length: 60) %></p>
+              <p class="mt-4 text-lg font-semibold text-gray-900">$<%= product.price %></p>
+              <p class="mt-4 text-lg font-medium text-gray-900">Stock: <%= product.stock.presence || "N/A" %></p>
+            </div>
+          
+            <div class="absolute inset-0 z-10 flex items-center justify-center bg-gray-800 bg-opacity-75 text-white opacity-0 group-hover:opacity-100 transition-opacity">
+              <a href="<%= product_path(product) %>" class="text-center text-sm font-medium px-4 py-2 bg-indigo-600 hover:bg-indigo-700 rounded-md">
+                View Details
+              </a>
+            </div>
+          </div>
         <% end %>
-      </tbody>
-    </table>
+      </div>
+    </div>
   </div>
+<% else %>
```

Comment:
Since you've already assigned the record in your index action, there's no need for the if-else block. Simply use @product as mentioned above in the view without the condition.